lazy_load_symbols/load_sym_lazily.py

```diff
@@ -0,0 +1,255 @@
+    '''
+
+    def __init__(self):
+        super(ReverseStepSymbol, self).__init__('rss', gdb.COMPLETE_EXPRESSION)
+
+    def get_frame_name(self, pc):
+        sodict = load_so_list()
+        loaded = False
+        try:
+            for name in sodict:
+                sa = int(sodict[name].start, 16)
+                ea = int(sodict[name].end, 16)
+                if pc >= sa and pc < ea:
+                    gdb.execute('sharedlibrary {}'.format(name))
+                    print('found sharedlibrary', name)
+                    loaded = True
+            if not loaded:
+                print 'ERROR: shared library not found for address {pc}'
+        except gdb.error as e:
+            print('ERROR: gdb eror', e)
+
+    def invoke(self, arg, from_tty):
+        #gdb.execute('bt 1')
+        gdb.execute('rsi', to_string=True)  # hide the output
+        #gdb.execute('bt 1')
+        # code to resolve the symbol at pc address
+        curr_frame = gdb.newest_frame()
+        print(curr_frame, "PC", curr_frame.pc(), "name", curr_frame.name())
+        if not curr_frame.name():
+            self.get_frame_name(curr_frame.pc())
+        try:
+            curr_frame = gdb.newest_frame()
+            print(curr_frame.name())
+        except gdb.error as e:
+            print('ERROR:', e)
+        # check symbol is now present
+        hex_pc = hex(curr_frame.pc())
+        if not curr_frame.name():
+            print('WARNING: reverse step may be slow, no symbol found for PC {}'.format(hex_pc))
+        gdb.execute('si', to_string=True)  # hide the output
+        time_gdb_execute('rs')
+        gdb.execute('bt 1')
+
+class LoadLibFromSym(gdb.Command):
+    '''
+    Command to load all libs that might define symbol
+    '''
+
+    def __init__(self):
+        super(LoadLibFromSym, self).__init__('loadsymlib', gdb.COMPLETE_EXPRESSION)
+        self.sodict = {}
+
+    def read_lib_symbols(self, so):
+        fn = self.sodict[so].fullname
+        rs = parse_raw_symlist(extract_symbol_list_from_so(fn))
+        self.sodict[so]._replace(symbols = extract_symbols_from_raw_list(rs))
+
+    def find_lib_with_symbol(self, symbol):
+        for so, content in self.sodict.items():
+            if not content.loaded:
+                if not content.symbols:
+                    self.read_lib_symbols(so)
+                if symbol in content.symbols:
+                    content.loaded = True
+                    return so
+        return None
+
+    def load_libs_with_name(self, symbol):
+        match = self.find_lib_with_symbol(symbol)
+        assert match, 'didn\'t find a library with symbol {}'.format(symbol)
+        gdb.execute('sharedlibrary {}'.format(match))
+
+    def invoke(self, symbol, from_tty):
+        try:
+            if not self.sodict:
+                self.sodict = load_so_list()
+            self.load_libs_with_name(symbol)
+        except Exception:
+            traceback.print_exc()
```

Comment:
You don't need to do this.

Reply:
I do, I wasn't able to see the traceback otherwise